homeassistant/components/tuya/button.py

```diff
@@ -98,8 +143,15 @@ def __init__(
         """Init Tuya button."""
         super().__init__(device, device_manager)
         self.entity_description = description
-        self._attr_unique_id = f"{super().unique_id}{description.key}"
+        self._attr_unique_id = f"{super().unique_id}{description.key}{description.name}"
```

Comment:
Why do we append the name here? Feels a bit weird to make the name part of a unique key.

Reply:
Maybe it wasn't a good idea to use the name key, but I did it to make it possible to create more than one button with the same key, for example to make a select that might not have good usability for the user in buttons.